File: weather-mvp-dagger/src/main/java/com/baronzhang/android/weather/view/selectcity/SelectCityComponent.java
package com.baronzhang.android.weather.view.selectcity;

import com.baronzhang.android.weather.di.ApplicationComponent;
import com.baronzhang.android.weather.di.ActivityScoped;

import dagger.Component;

/**
 * @author baronzhang (baron[dot]zhanglei[at]gmail[dot]com)
 *         2016/11/30
 */
@ActivityScoped
@Component(modules = SelectCityModule.class, dependencies = ApplicationComponent.class)
public interface SelectCityComponent {

    void inject(SelectCityActivity selectCityActivity);
}

File: weather-mvp-dagger-kotlin/src/main/java/com/minimalist/weather/kotlin/view/selectcity/SelectCityComponent.kt
package com.minimalist.weather.kotlin.view.selectcity


import com.minimalist.weather.kotlin.di.ActivityScoped
import com.minimalist.weather.kotlin.di.ApplicationComponent

import dagger.Component

@ActivityScoped
@Component(modules = arrayOf(SelectCityModule::class), dependencies = arrayOf(ApplicationComponent::class))
interface SelectCityComponent {

    fun inject(selectCityActivity: SelectCityActivity)
}
